projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loans.component.ts

```diff
@@ -65,31 +69,30 @@ export class PatronProfileLoansComponent implements OnInit, OnDestroy {
       );
     this._subscription.add(
       this._paginator.more$.subscribe((page: number) => {
-        this._loanQuery(page).subscribe((response: Record) => {
-          this.records = this.records.concat(response.hits.hits);
-        });
+        this.records = this._sortedLoans();
       })
     );
     this._subscription.add(
       this._patronProfileMenuService.onChange$.subscribe(() => {
-        this._initialLoad();
+        this._initialLoadSorted();
       })
     );
-    this._loanQuery(1).subscribe((response: Record) => {
-      this._initialLoad();
-    });
+    this._initialLoadSorted();
   }
 
   /** OnDestroy hook */
   ngOnDestroy(): void {
     this._subscription.unsubscribe();
   }
 
-  /** Initial records load */
-  private _initialLoad(): void {
-    this._loanQuery(1).subscribe((response: Record) => {
-      this._paginator.setRecordsCount(response.hits.total.value);
-      this.records = response.hits.hits;
+  /** Initial records load and sort */
+  private _initialLoadSorted(): void {
+    const pgr = this._paginator;
+    this._loanQueryAll().subscribe((response: Record) => {
```

Comment:
It is not a good idea to load all the loans at once, because if there are a lot of them, it will make the thing explode. The pager is there to avoid this.

You really need to use the pager.